tests/lib/rules/max-lines.js

```diff
@@ -147,17 +194,243 @@ ruleTester.run("max-lines", rule, {
+                    line: 3,
+
+                    endLine: 5
+
+                }
+            ]
+        },
+        {
+            code: [
+                "var a = 'a'; ",
+                "var x",
+                "var c;",
+                "console.log",
+                "/** block \n\n comments */"
+            ].join("\n"),
+            options: [{ max: 2, skipComments: true }],
+            errors: [
+                {
+                    messageId: "exceed",
+                    data: { max: 2, actual: 4 },
+                    line: 3,
+
+                    endLine: 7
+
+                }
+            ]
+        },
+        {
+            code: [
+                "var a = 'a'; ",
+                "var x",
+                "\n",
+                "var c;",
+                "console.log",
+                "\n"
+            ].join("\n"),
+            options: [{ max: 2, skipBlankLines: true }],
+            errors: [
+                {
+                    messageId: "exceed",
+                    data: { max: 2, actual: 4 },
+                    line: 5,
+
+                    endLine: 8
+
+                }
+            ]
+        },
+        {
+            code: [
+                "var a = 'a'; ",
+                "\n",
+                "var x",
+                "var c;",
+                "console.log",
+                "\n"
+            ].join("\n"),
+            options: [{ max: 2, skipBlankLines: true }],
+            errors: [
+                {
+                    messageId: "exceed",
+                    data: { max: 2, actual: 4 },
+                    line: 5,
+
+                    endLine: 8
+
+                }
+            ]
+        },
+        {
+            code: [
+                "var a = 'a'; ",
+                "//",
+                "var x",
+                "var c;",
+                "console.log",
+                "//"
+            ].join("\n"),
+            options: [{ max: 2, skipComments: true }],
+            errors: [
+                {
+                    messageId: "exceed",
+                    data: { max: 2, actual: 4 },
+                    line: 4,
+
+                    endLine: 6
+
+                }
+            ]
+        },
+        {
+            code: ["// hello world", "/*hello", " world 2 */", "var a,", "b", "// hh", "c,", "e,", "f;"].join("\n"),
+            options: [{ max: 2, skipComments: true }],
+            errors: [{
+                line: 7,
+
+                data: { max: 2, actual: 5 },
+                messageId: "exceed",
+                endLine: 9
+
+            }]
+        },
+        {
+            code: ["", "var x = '';", "", "// comment", "", "var b = '',", "c,", "d,", "e", "", "// comment"].join("\n"),
+            options: [{ max: 2, skipComments: true, skipBlankLines: true }],
+            errors: [{
+                data: { max: 2, actual: 5 },
+                messageId: "exceed",
+                line: 7,
+                endLine: 11
+            }]
```

Comment:
It might be nice to have error properties ordered like in all other tests above.